src/taskgraph/main.py

```diff
@@ -862,6 +862,8 @@ def init_taskgraph(options):
         return 1
 
     context["repo_name"] = urlparse(repo_url).path.rsplit("/", 1)[-1]
+    if context["repo_name"].endswith(".git"):
```

Comment:
Improvement for another time: we could likely factor out this special case and use https://github.com/mozilla-releng/mozilla-repo-urls/. No need to handle this now, though.